src/braket/default_simulator/openqasm/interpreter.py

```diff
@@ -309,7 +309,12 @@ def inline_gate_def_body(self, body: List[QuantumStatement]) -> List[QuantumStat
                 statement.arguments = self.visit(statement.arguments)
                 statement.modifiers = self.visit(statement.modifiers)
                 statement.qubits = self.visit(statement.qubits)
-                if gate_name in BRAKET_GATES:
+                try:
+                    self.context.get_gate_definition(gate_name)
+                    user_defined_gate = True
+                except ValueError:
+                    user_defined_gate = False
+                if gate_name in BRAKET_GATES and not user_defined_gate:
```

Comment:
nit: This condition is used below too, so it can be pulled into a private helper

Reply:
Ya I considered it, but seemed small enough to leave out. Happy to factor out if you think that'd be better